src/query/sql/src/planner/plans/scalar_expr.rs

```diff
@@ -82,7 +82,9 @@ impl PartialEq for ScalarExpr {
     #[recursive::recursive]
     fn eq(&self, other: &Self) -> bool {
         match (self, other) {
-            (ScalarExpr::BoundColumnRef(l), ScalarExpr::BoundColumnRef(r)) => l.eq(r),
+            (ScalarExpr::BoundColumnRef(l), ScalarExpr::BoundColumnRef(r)) => {
+                l.column.index == r.column.index && l.column.table_index == r.column.table_index
```

Comment:
if index is `DUMMY_INDEX`, name should be compared.

Reply:
`DUMMY_COLUMN_INDEX` or `DUMMY_INDEX` will not appear in `ScalarExpr::BoundColumnRef`.